### PR TITLE
Parse decimal units

### LINKED ISSUE #2516: Durations should allow for decimals
**Describe the bug**
`10.5sec` should parse but it's not currently allowed.

**To Reproduce**
1. Use a decimal with a unit, like `10.5sec`

**Expected behavior**
This is a handy shorthand rather than having to write out the a decimal + unit. Some of these don't make sense, since `0.5ns` isn't currently supported, so we'd need to figure out how we want to do it.

**Configuration (please complete the following information):**
 - OS (e.g. Windows): Any
 - Nu version (you can use the `version` command to find out): latest main

### PR DESCRIPTION
**Pull request by luccasmmg**
Fixes #2516 

This solves the issue that jonathan pointed out about, and also adds the possibility of parcing decimal filesizes as well, the units are converted to what makes most sense(Ex: 1.5min is converted to 90 seconds or 1.5mb is converted to 1500kb) i wantedget the opinions of other people about these conversions(Ex: if one day should be converted to 1440 minutes or 24 hours), below are some examples that i did. 
image
image
image

### CODE REVIEW

**Comment by ilius** [comment]
I think there is a bug

```
(nu) nushell $ = 10kib
10.2 KB
(nu) nushell $ = 10.0kib
10.0 EB
```

File: crates/nu-parser/src/parse.rs [hunk 1]
```diff
@@ -325,51 +325,74 @@ fn parse_operator(lite_arg: &Spanned<String>) -> (SpannedExpression, Option<Pars
...
-        (Unit::Day, vec!["day"]),
-        (Unit::Week, vec!["wk"]),
-        (Unit::Month, vec!["mon"]),
-        (Unit::Year, vec!["yr"]),
+        (Unit::Kilobyte, "KB", Some((Unit::Byte, 1000))),
+        (Unit::Megabyte, "MB", Some((Unit::Kilobyte, 1000))),
+        (Unit::Gigabyte, "GB", Some((Unit::Megabyte, 1000))),
+        (Unit::Terabyte, "TB", Some((Unit::Gigabyte, 1000))),
+        (Unit::Petabyte, "PB", Some((Unit::Terabyte, 1000))),
+        (Unit::Kibibyte, "KIB", Some((Unit::Petabyte, 1000))),
```
Comment (ilius) [inline crates/nu-parser/src/parse.rs:341]: These 3 are wrong, shoulf be
```
        (Unit::Kibibyte, "KIB", Some((Unit::Byte, 1024))),
        (Unit::Mebibyte, "MIB", Some((Unit::Kibibyte, 1024))),
        (Unit::Gibibyte, "GIB", Some((Unit::Mebibyte, 1024))),
```

File: crates/nu-parser/src/parse.rs [hunk 2]
```diff
@@ -2179,3 +2202,79 @@ fn unit_parse_byte_units() {
...
+            string: String::from("11.5Tb"),
+            value: 11500,
+            value_str: String::from("11.5"),
+            unit: Unit::Gigabyte,
+        },
+        TestCase {
+            string: String::from("10.5kib"),
+            value: 10500,
+            value_str: String::from("10.5"),
+            unit: Unit::Petabyte,
```
Comment (ilius) [inline crates/nu-parser/src/parse.rs:2244]: Kibibyte

File: crates/nu-parser/src/parse.rs [hunk 3]
```diff
@@ -2179,3 +2202,79 @@ fn unit_parse_byte_units() {
...
+        },
+        TestCase {
+            string: String::from("0.5mib"),
+            value: 500,
+            value_str: String::from("0.5"),
+            unit: Unit::Kibibyte,
```
Comment (ilius) [inline crates/nu-parser/src/parse.rs:2250]: 0.5 Mebibyte or 512 Kibibyte

File: crates/nu-parser/src/parse.rs [hunk 4]
```diff
@@ -2179,3 +2202,79 @@ fn unit_parse_byte_units() {
...
+        },
+        TestCase {
+            string: String::from("3.25gib"),
+            value: 3250,
+            value_str: String::from("3.25"),
+            unit: Unit::Mebibyte,
```
Comment (ilius) [inline crates/nu-parser/src/parse.rs:2256]: Gibibyte

**Comment by luccasmmg** [comment]
@ilius thanks for the input, as soon as i get home i will fix these tests, tried to make a quick fix and end up forgetting to check the tests

**Comment by jonathandturner** [comment]
Looks good!

**Pull request merged**